src/CSharp/CSharp/CSharpUtility.cs

```diff
@@ -681,4 +681,11 @@ public static SeparatedSyntaxList<TypeParameterSyntax> GetTypeParameters(SyntaxN
                 return null;
         }
     }
+
+    public static bool IsEnumMemberObsolete(EnumMemberDeclarationSyntax enumMember)
+    {
+        return enumMember.AttributeLists
+            .SelectMany(attrList => attrList.Attributes)
+            .Any(attribute => attribute.Name.ToString() == "Obsolete");
+    }
```

Comment:
This implementation probably would work for most of cases but I believe it could be improved.

It's better to use a symbol and check if it has an attribute. The symbol variable is already present in the analyzer and also in the code fix provider so this should be easy.

The method to be used is `symbol.HasAttribute(MetadataName)` and static class `MetadataNames` already contains member `MetadataNames.System_ObsoleteAttribute`.

As a result the method will be more general as it will check if a symbol is marked with `Obsolete` attribute.

Reply:
Thanks for the suggestion. I've updated the code to use `symbol.HasAttribute(MetadataNames.System_ObsoleteAttribute)` as recommended. I really appreciate your feedback.